db/db_ut.c

```diff
@@ -416,46 +416,35 @@ int db_print_set(const db_con_t* _c, char* _b, const int _l, const db_key_t* _k,
 int db_is_neq_type(db_type_t _t0, db_type_t _t1)
 {
 	// LM_DBG("t0=%d t1=%d!\n", _t0, _t1);
-	if(_t0 == _t1)
-		return 0;
-
-	switch(_t1) {
-	case DB_INT:
-		if(_t0==DB_BIGINT || _t0==DB_DATETIME || _t0==DB_BITMAP)
-			return 0;
-		break;
-
-	case DB_STRING:
-		if(_t0==DB_STR || _t0 == DB_BLOB)
-			return 0;
-		break;
-
-	case DB_STR:
-		if(_t0==DB_STRING || _t0==DB_BLOB)
-			return 0;
-		break;
-
+	switch(_t0) {
 	case DB_BIGINT:
-		if(_t0==DB_INT || _t0==DB_DATETIME || _t0==DB_BITMAP)
-			return 0;
-		break;
-
-	case DB_BLOB:
-		if(_t0==DB_STR || _t0==DB_STRING)
-			return 0;
-		break;
-
+	case DB_BITMAP:
 	case DB_DATETIME:
-		if(_t0==DB_INT || _t0==DB_BIGINT || _t0==DB_BITMAP)
+	case DB_INT:
+		switch(_t1) {
+		case DB_BIGINT:
+		case DB_BITMAP:
+		case DB_DATETIME:
+		case DB_INT:
 			return 0;
-		break;
-
-	case DB_BITMAP:
-		if(_t0==DB_INT || _t0==DB_BIGINT || _t0==DB_DATETIME)
+		default:
+			break;
+		}
+	case DB_BLOB:
```

Comment:
Optimization quirk: shouldn't there be a `break` after this curly brace?  E.g., on a `DB_INT` vs. `DB_BLOB` comparison, `_t0` will continue to fall through the BLOB, STR, STRING before finally exiting, rather than doing it immediately.

Reply:
Ah. Good call. I mistakenly thought I was breaking from _t0 in default, but I'm breaking from _t1.

Fixing...
```c
...
                case DB_INT:
                        return 0;
                default:
                        break;
                }       
        case DB_BLOB:
```
->
```c
...
                case DB_INT:
                        return 0;
                default: // still needs this, to avoid complaints from compiler
                        break;
                }
                break;
        case DB_BLOB:
```
->
```c
...
                case DB_INT:
                        return 0;
                default:
                        return 1;
                }
        case DB_BLOB:
```
Chose the last one then.